component-runtime-manager/src/test/java/org/talend/sdk/component/runtime/manager/asm/PluginGenerator.java

```diff
@@ -73,23 +75,35 @@ public File createChainPlugin(final File dir, final String plugin, final int max
             ofNullable(root.listFiles())
                     .map(Stream::of)
                     .orElseGet(Stream::empty)
-                    .filter(c -> c.getName().endsWith(".class"))
+                    .filter(c -> c.getName().endsWith(".class") || c.getName().endsWith(".json")
+                            || c.getName().endsWith(".csv"))
                     .forEach(clazz -> {
-                        try (final InputStream is = new FileInputStream(clazz)) {
-                            final ClassReader reader = new ClassReader(is);
-                            final ClassWriter writer = new ClassWriter(COMPUTE_FRAMES);
-                            reader.accept(new ClassRemapper(writer, new Remapper() {
-
-                                @Override
-                                public String map(final String key) {
-                                    return key.replace(sourcePackage, toPack).replace(fromPack, packageName);
-                                }
-                            }), EXPAND_FRAMES);
-                            outputStream.putNextEntry(new JarEntry(toPack + '/' + clazz.getName()));
-                            outputStream.write(writer.toByteArray());
-                            outputStream.closeEntry();
-                        } catch (final IOException e) {
-                            fail(e.getMessage());
+                        if (clazz.getName().endsWith(".class")) {
+                            try (final InputStream is = new FileInputStream(clazz)) {
+                                final ClassReader reader = new ClassReader(is);
+                                final ClassWriter writer = new ClassWriter(COMPUTE_FRAMES);
+                                reader.accept(new ClassRemapper(writer, new Remapper() {
+
+                                    @Override
+                                    public String map(final String key) {
+                                        return key.replace(sourcePackage, toPack).replace(fromPack, packageName);
+                                    }
+                                }), EXPAND_FRAMES);
+                                outputStream.putNextEntry(new JarEntry(toPack + '/' + clazz.getName()));
+                                outputStream.write(writer.toByteArray());
+                                outputStream.closeEntry();
+                            } catch (final IOException e) {
+                                fail(e.getMessage());
+                            }
+                        } else {
+                            try {
+                                final FileReader reader = new FileReader(clazz);
```

Comment:
I'm trying to find in the ticket an explanation why do we want to define classes as .json or .csv files.
It seems very strange and confusing